src/test/java/uk/gov/hmcts/dts/fact/services/MapitServiceTest.java

```diff
@@ -124,4 +129,42 @@ void shouldReturnOptionalEmptyIfBlankPostcodeForPartial() {
         assertThat(result).isNotPresent();
         verifyNoInteractions(mapitClient);
     }
+
+    @Test
+    void localAuthorityExistsShouldReturnTrueForValidLocalAuthorityNames() {
+        LinkedHashMap<String, MapitArea> mockAreaData = new LinkedHashMap<>();
+        mockAreaData.put("100", new MapitArea("100", "Birmingham City Council", "MTD"));
+        when(mapitClient.getMapitDataForLocalAuthorities(any())).thenReturn(mockAreaData);
+
+        assertThat(mapitService.localAuthorityExists("Birmingham City Council")).isTrue();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "", " " })
+    @NullSource
+    void localAuthorityExistsShouldReturnFalseForEmptyLocalAuthorityNames(String name) {
+        assertThat(mapitService.localAuthorityExists(name)).isFalse();
+
+        // We don't need to call the client to validate the name if the name is
+        // null, empty or whitespace
+        verify(mapitClient, never()).getMapitDataForLocalAuthorities(any());
```

Comment:
Nothing wrong with this line. Just let you know if there is no interaction with a class, we can also do the following instead:
verifyNoInteractions(mapItClient)